controllers/akodeploymentconfig/akodeploymentconfig_controller_cluster_phase.go

```diff
@@ -93,12 +91,9 @@ func (r *AKODeploymentConfigReconciler) applyClusterLabel(
 	} else {
 		log.Info("Label already applied to cluster", "label", akoov1alpha1.AviClusterLabel)
 	}
-	selector, err := metav1.LabelSelectorAsSelector(&obj.Spec.ClusterSelector)
-	if err != nil {
-		return ctrl.Result{}, err
-	}
-	// cluster selected by AKODeploymentConfig with selectors
-	if !selector.Empty() {
+	// When the cluster is selected by this AKODeploymentConfig, but its is not install-ako-for-all,
+	// this indicates that the cluster is managed by a customized ADC and the default ADC shall be skipped.
+	if obj.Name != akoov1alpha1.WorkloadClusterAkoDeploymentConfig {
 		cluster.Labels[akoov1alpha1.AviClusterSelectedLabel] = ""
```

Comment:
Is this label applicable to workload-cluster only?

Reply:
from my understanding. no. it should be apply to all.

the management cluster will be selected by `install-ako-for-all` and `install-ako-for-mgmt` simultaneously. It is expected to be labelled with `networking.tkg.tanzu.vmware.com/avi-skip-default-adc` as well

Reply:
Then I think we should check the cluster type here. If workload-cluster, put label only if adc is not `install-ako-for-all`. If mgmt-cluster, put label only if adc is not `install-ako-for-mgmt-cluster`

Reply:
no need to check the cluster type here. We can treat mc the same way as other workload cluster. So normally mc will have the label since `install-ako-for-mgmt` is treated as a customized ADC. @sbhatp please let me know if you still have any concern with this logic.

Reply:
Got it. Since `install-ako-for-mgmt` has `clusterSelector`, its not a match-all adc like `install-ako-for-all`